src/Cupboard.Providers.Windows/Chocolatey/ChocolateyPackageProvider.cs

```diff
@@ -28,12 +28,42 @@ public override ChocolateyPackage Create(string name)
 
         protected override bool IsPackageInstalled(ChocolateyPackage resource, string output)
         {
-            return output.Contains(resource.Package, StringComparison.OrdinalIgnoreCase);
+            var outputSpan = output.AsSpan();
+            var containsPackage = outputSpan.Contains(resource.Package.AsSpan(), StringComparison.OrdinalIgnoreCase);
+            if (string.IsNullOrEmpty(resource.PackageVersion))
+            {
+                return containsPackage;
+            }
+
+            if (!containsPackage)
+            {
+                return false;
+            }
+
+            var indexOfPipe = outputSpan.IndexOf('|');
+            if (indexOfPipe == -1)
+            {
+                return false;
+            }
+
+            var indexOfPipePlusOne = indexOfPipe + 1;
+            var indexOfNewLine = outputSpan[indexOfPipePlusOne..].IndexOfAny('\r', '\n');
+            if (indexOfNewLine == -1)
+            {
+                return false;
+            }
+
+            var versionSpan = outputSpan.Slice(indexOfPipePlusOne, indexOfNewLine);
+            return Version.TryParse(versionSpan, out var currentPackageVersion)
+                   && Version.TryParse(resource.PackageVersion.AsSpan(), out var packageVersion)
+                   && ((resource.AllowDowngrade is true && currentPackageVersion == packageVersion)
+                       || (resource.AllowDowngrade is false && currentPackageVersion >= packageVersion));
```

Comment:
@patriksvensson tried to implement allow downgrade.
Not sure if this chain of condition is too much or would you like it to be simplified.

Reply:
We can simplify it in a future PR. No worries.